packages/beacon-node/src/chain/historicalState/worker.ts

```diff
@@ -0,0 +1,176 @@
+import worker from "node:worker_threads";
+import {Transfer, expose} from "@chainsafe/threads/worker";
+import {createBeaconConfig, chainConfigFromJson} from "@lodestar/config";
+import {getNodeLogger} from "@lodestar/logger/node";
+import {
+  BeaconStateTransitionMetrics,
+  EpochTransitionStep,
+  PubkeyIndexMap,
+  StateCloneSource,
+  StateHashTreeRootSource,
+} from "@lodestar/state-transition";
+import {LevelDbController} from "@lodestar/db";
+import {RegistryMetricCreator, collectNodeJSMetrics} from "../../metrics/index.js";
+import {JobFnQueue} from "../../util/queue/fnQueue.js";
+import {QueueMetrics} from "../../util/queue/options.js";
+import {BeaconDb} from "../../db/index.js";
+import {HistoricalStateWorkerApi, HistoricalStateWorkerData} from "./types.js";
+import {getHistoricalState} from "./getHistoricalState.js";
+
+// most of this setup copied from networkCoreWorker.ts
+
+const workerData = worker.workerData as HistoricalStateWorkerData;
+
+// TODO: Pass options from main thread for logging
+// TODO: Logging won't be visible in file loggers
```

Comment:
As in we will not see this in our logs on hetzner?

Reply:
this is copy/pasta from network/discv5